doc/threat_model/TBF_Headers.md

```diff
@@ -0,0 +1,16 @@
+TBF Headers
+===========
+
+TBF headers are considered part of an application, and are similarly untrusted.
+As such, TBF header parsing must be robust against malicious inputs (e.g.
+pointers must be checked to confirm they are in-bounds for the app).
+
+However, because the kernel relies on the TBF's `total_size` field to load the
+applications, the application loader is responsible for verifying the
+`total_size` field at install time.
+
+When possible, [TLV types](../TockBinaryFormat.md#tlv-types) should be designed
+so that the kernel does not need to trust their correctness. When a TLV type is
+defined that the kernel must trust, then the threat model must be updated to
+indicate that application loaders are responsible for verifying the value of
+that TLV type.
```

Comment:
I don't think so. An application cannot impact another application by tampering with its own `init_offset` or `protected_size`, only cause itself to fail.

That said, I think `minimum_ram_size` is of concern. I think the application loader needs to check that, but I'm not 100% sure that doing so is feasible.

Question for reviewers more familiar with how tockloader works: is it feasible for tockloader to verify that a set of applications fits within the available RAM space on a board?

Reply:
Tockloader is entirely in service of support tock and making it easy to use. So if we are concerned about apps fitting then tockloader should absolutely help with that.

The challenge for doing that, however, is mimicking the memory allocation policy of different architectures in tockloader. Since how much memory is given to an app is a function of the app's request and the MPU trait, tockloader would have to know that algorithm.

Reply:
> The challenge for doing that, however, is mimicking the memory allocation policy of different architectures in tockloader.

I didn't think about that -- tockloader doesn't currently have that information. That throws a wrench into the works.

We've been tossing around ideas for using static linking to solve Rust and RISC-V's relocation problems. It seems to me that implementing one of those static linking ideas would solve this particular problem. It seems convenient that one piece of design/engineering work would solve two problems, but I'm a bit hesitant to mix relocation work with threat model work.

Reply:
We discussed this during the core working group call today. We are removing application loaders from the TCB for availability of applications by removing the guarantee that Tock will launch all available applications.

The application loader will still need to check `total_size` for confidentiality and integrity reasons.